ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -26,19 +26,23 @@ internal sealed class OfferingSelection {
 class PaywallOptions(builder: Builder) {
 
     internal val offeringSelection: OfferingSelection
-    val shouldDisplayDismissButton: Boolean
+    private val shouldDisplayDismissButton: Boolean
     val fontProvider: FontProvider?
     val listener: PaywallListener?
     internal var mode: PaywallMode = PaywallMode.default
+    val dismissRequest: () -> Unit
 
     init {
         this.offeringSelection = builder.offeringSelection
         this.shouldDisplayDismissButton = builder.shouldDisplayDismissButton
         this.fontProvider = builder.fontProvider
         this.listener = builder.listener
+        this.dismissRequest = builder.dismissRequest
     }
 
-    class Builder {
+    class Builder(
+        internal val dismissRequest: () -> Unit,
```

Comment:
Considering this is now mandatory, I think we should remove the `onCloseButtonPressed` on the `PaywallListener`. Wdyt?

Reply:
I agree, I think having both is confusing

Reply:
Ooh I missed that yeah